src/views/documents/utils/QuerySortDropdown.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <dropdown-button>
+    <span slot="button" class="button is-small-mobile" :disabled="false">
+      <fa-icon icon="sort" />
+      <span class="is-hidden-mobile"> &nbsp;{{ $gettext('Sort by') }} </span>
+      <!-- <span v-if="category.activeCount != 0"> &nbsp;({{ category.activeCount }}) </span> -->
+      <span>&nbsp;</span>
+      <fa-icon icon="angle-down" aria-hidden="true" />
+    </span>
+
+    <div v-if="complexSort" class="content">
+      <div class="dropdown-item is-italic" v-translate>You are using a custom sort</div>
+      <div class="dropdown-item">
+        <button class="button" v-translate @click="clear">Reset</button>
+      </div>
+    </div>
+    <div v-else-if="sortableFields.length !== 0" class="content">
+      <div class="dropdown-item sort-item" v-for="field of sortableFields" :key="field">
+        <span
+          class="has-cursor-pointer is-nowrap sort-label"
+          :class="{ 'has-text-weight-bold': field == sortField }"
+          @click="toggle(field)"
+        >
+          {{ $gettext(field) }}
+        </span>
+        <span v-if="field == sortField" class="has-cursor-pointer">
+          <fa-icon v-if="reversed" icon="sort-amount-up" @click="setSort(field)" />
+          <fa-icon v-if="!reversed" icon="sort-amount-down-alt" @click="setSort('-' + field)" />
+        </span>
+      </div>
+    </div>
+    <div v-else class="content">
+      <div class="dropdown-item is-italic">
+        <span v-if="documentType === 'waypoint'" v-translate> Please set a waypoint type to enable sort feature </span>
```

Comment:
I don't understand this one either

Reply:
On waypoints, fields are visible/hidden, depending the waypoint's type.